application/electron/src/services/files/service.file.opener.ts

```diff
@@ -228,6 +256,94 @@ class ServiceFileOpener implements IService {
+                                        lastModified: fsStats.mtimeMs,
+                                        lastModifiedDate: fsStats.mtime,
+                                        name: self._getName(file),
+                                        path: file,
+                                        size: fsStats.size,
+                                        type: 'file',
+                                    }));
+                                }
+                            });
+                        } else if (!(lsStats.isDirectory())) {
+                            // Neither file nor directory
+                            const errorMessage = `Fail to get file info of ${file} because it is neither a file nor a directory`;
+                            ServiceNotifications.notify({
+                                caption: `Error with file`,
+                                message: errorMessage,
+                                type: ENotificationType.warning,
+                            });
+                            rejected(new Error(errorMessage));
+                        } else {
+                            // Directory
+                            return fs.readdir(file, (err, files) => {
+                                if (err) {
+                                    const errorMessage = `Fail to list files of directory ${file} due to error: ${err.message}`;
+                                    ServiceNotifications.notify({
+                                        caption: 'Error with directory',
+                                        message: errorMessage,
+                                        type: ENotificationType.warning,
+                                    });
+                                    rejected(new Error(errorMessage));
+                                } else {
+                                    Promise.all(files.map((subFile: string) => {
+                                        return listAllFiles(file + '/' + subFile);
+                                    })).then(() => {
+                                        resolve(allFiles);
+                                    }).catch((error: Error) => {
+                                        rejected(new Error(`Fail to list files of directory ${file} info due to error: ${error.message}`));
+                                    });
+                                }
+                            });
+                        }
+                    });
+                });
+            }
+            listAllFiles(startFile).then(() => {
+                resolve(allFiles);
+            }).catch((error: Error) => {
+                reject(new Error(`Fail to list files of directory ${startFile} info due to error: ${error.message}`));
+            });
+        });
+    }
+
+    private _getName(file: string): string {
+        if (file.trim() === '') {
+            return '';
+        }
+        return path.basename(file);
+    }
```

Comment:
Do we really need this method? Probably we can just call `path.basename` where it would be needed?